gridcomps/History/MAPL_HistoryGridComp.F90

```diff
@@ -3596,6 +3596,11 @@ subroutine Run ( gc, import, export, clock, rc )
          call list(n)%trajectory%regrid_accumulate(_RC)
          if( ESMF_AlarmIsRinging ( list(n)%trajectory%alarm ) ) then
             call list(n)%trajectory%append_file(current_time,_RC)
+            !!bug
+            !!the barrier did not work
+            call ESMF_GridCompGet(gc, vm=vm, _RC)
+            call ESMF_VMGetCurrent(vm, _RC)
+            call ESMF_VMbarrier(vm, _RC)
```

Comment:
I realized the usage of the ESMF_VMbarrier is right. I was complaining about this, because after this barrier step, the generated netCDF files by call append_file shows 2.1 MB and all the fields are there, except the 2d and 3d fields are zero.  I was expecting a full content in netCDF, but it does not happen.  So it is a separate problem.  

Maybe after
`
if (mapl_am_i_root()) then
 call this%file_handle%put_var(trim(item%xname),p_acc_rt_3d(:,:), start=[is,1],count=[nx,size(p_acc_rt_3d,2)])
endif
`
I should close the NC file on root.  Then I would get a finished NC file at the barrier point.
I will remove the comment. (So it is a bug)

Reply:
Yes - I don't think NetCDF guarantees anything until the file is closed.  But I'm not an expert.

Reply:
@tclune  Yes! you are right ---  the  netCDF put_var does not mean it is writing to disk.  
I tested 
``` 
            call list(n)%trajectory%append_file(current_time,_RC)
            call list(n)%trajectory%close_file_handle(_RC)
            _FAIL('test close file')
```  
Now I immediately get the correct content 2d and 3d in NC files.  So that is the final for netCDF write to disk.

Reply:
I still see some netCDF latency phenomena on Mac that I donot understand as I am not an expert on netCDF.  But it does not affect the results and performance of the code, so I am fine with it. I will update the PR. Thanks.